### PR TITLE
AKS Endpoint Caller

### PR DESCRIPTION
**Pull request by v-fearam**
It generates the first app version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,81 @@
...
+  template:
+    metadata:
+      labels:
+        app: ingress-endpoint
+    spec:
+      containers:
+      - name: ingress-endpoint
+        image: myacr.azurecr.io/aks-endpoint-caller
+        ports:
+        - containerPort: 80
```
Comment (ckittel) [inline README.md:47]: Let's keep it exposed on 8080 (adding the `http://+:8080` environment variable)

File: README.md [hunk 2]
```diff
@@ -1 +1,81 @@
...
+      app: ingress-endpoint
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: ingress-endpoint
+    spec:
+      containers:
+      - name: ingress-endpoint
+        image: myacr.azurecr.io/aks-endpoint-caller
```
Comment (ckittel) [inline README.md:45]:
```suggestion
        image: myacr.azurecr.io/aks-endpoint-caller:1.0
```

File: README.md [hunk 3]
```diff
@@ -1 +1,81 @@
...
+          value: "http://service-a,http://service-b,http://service-c"
+        - name: EXTERNAL_DEPENDENCIES
+          value: "https://whatismyip.io/"
+        - name: "DEPTH"
+          value: "2"
+```
+We need to define some environment variables:
+1. EXTERNAL_DEPENDENCIES, These are URLs with an unknown result
+1. SELF_HOSTS_DEPENDENCIES, These are URLs where instances of this micro-service is deployed, so the result format is well known
+1. DEPTH, This value is for doing the recursive calls. The default is 0, not dependency is call. If the value is bigger than 0 the dependencies are call, and the recursive dependencies are called using (DEPTH-1). This parameter avoid infinite calls. This parameter is needed to be set only in the entry point of micro services components.
```
Comment (ckittel) [inline README.md:59]:
```suggestion
1. `DEPTH` - This value is for doing the recursive calls. The default is `0`, meaning no dependencies are called. If the value is larger than `0` the dependencies are call, and the recursive dependencies are called using (`DEPTH-1`). This parameter avoid infinite calls. This parameter is needed to be set only in the entry point of micro services components.
```

File: SimpleChainApi/SimpleChainApi/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#See https://aka.ms/containerfastmode to understand how Visual Studio uses this Dockerfile to build your images for faster debugging.
+
+FROM mcr.microsoft.com/dotnet/aspnet:5.0 AS base
+WORKDIR /app
+EXPOSE 80
```
Comment (ckittel) [inline SimpleChainApi/SimpleChainApi/Dockerfile:5]: Okay to just have this be 8080 to start with?  (I want to make sure we are working toward getting this to run without root -- it's not set up that way here, but I want to make sure that's the direction we're headed at least.)

File: SimpleChainApi/SimpleChainApi/Pages/Shared/_Layout.cshtml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        <link rel="stylesheet" href="~/css/site.css" />
+    </environment>
+    <environment exclude="Development">
+        <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css"
+              asp-fallback-href="~/lib/bootstrap/dist/css/bootstrap.min.css"
+              asp-fallback-test-class="sr-only" asp-fallback-test-property="position" asp-fallback-test-value="absolute" />
+        <link rel="stylesheet" href="~/css/site.min.css" asp-append-version="true" />
+    </environment>
+    <style>
+        .editor-label
```
Comment (ckittel) [inline SimpleChainApi/SimpleChainApi/Pages/Shared/_Layout.cshtml:19]: Do we know what these are for?

File: SimpleChainApi/SimpleChainApi/wwwroot/css/site.css [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    line-height: 1.4;
+}
+
+/* Make .svg files in the carousel display properly in older browsers */
+.carousel-inner .item img[src$=".svg"] {
+    width: 100%;
+}
+
+/* QR code generator */
+#qrCode {
```
Comment (ckittel) [inline SimpleChainApi/SimpleChainApi/wwwroot/css/site.css:27]: Do we know what this is for? And other items in here, like carousel.

File: README.md [hunk 7]
```diff
@@ -1 +1,81 @@
...
+        - name: SELF_HOSTS_DEPENDENCIES
+          value: "http://service-a,http://service-b,http://service-c"
+        - name: EXTERNAL_DEPENDENCIES
+          value: "https://whatismyip.io/"
+        - name: "DEPTH"
+          value: "2"
+```
+We need to define some environment variables:
+1. EXTERNAL_DEPENDENCIES, These are URLs with an unknown result
+1. SELF_HOSTS_DEPENDENCIES, These are URLs where instances of this micro-service is deployed, so the result format is well known
```
Comment (ckittel) [inline README.md:58]:
```suggestion
We need to define some environment variables:
1. `EXTERNAL_DEPENDENCIES` - These are URLs with an unknown result; as a comma-separated list
1. `SELF_HOSTS_DEPENDENCIES` - These are URLs where instances of this micro service is deployed, so the result format is well known; as a comma-separated list.
```

File: README.md [hunk 8]
```diff
@@ -1 +1,81 @@
...
+1. DEPTH, This value is for doing the recursive calls. The default is 0, not dependency is call. If the value is bigger than 0 the dependencies are call, and the recursive dependencies are called using (DEPTH-1). This parameter avoid infinite calls. This parameter is needed to be set only in the entry point of micro services components.
+
+### Example
+We can create the configuration which is useful for us. A possible setup is:
```
Comment (ckittel) [inline README.md:62]:
```suggestion
We can create the configuration which is useful for the situation we want to test. An example setup is:
```

File: README.md [hunk 9]
```diff
@@ -1 +1,81 @@
...
+1. Ingress Enpoint is the starting point, and calls: https://www.microsoft.com, https://serviceA, https://serviceB,  https://serviceC
+1.  serviceA calls: https://www.microsoft.com, https://serviceA, https://serviceB,  https://serviceC
+1.  serviceB calls: https://www.microsoft.com, https://serviceA, https://serviceB,  https://serviceC
+1.  serviceC calls: https://www.microsoft.com, https://serviceA, https://serviceB,  https://serviceC
```
Comment (ckittel) [inline README.md:66]:
```suggestion
1. Ingress Endpoint is the starting point, and calls: https://www.microsoft.com, https://serviceA, https://serviceB,  https://serviceC
1.  serviceA calls: https://www.microsoft.com, https://serviceA, https://serviceB, https://serviceC
1.  serviceB calls: https://www.microsoft.com, https://serviceA, https://serviceB, https://serviceC
1.  serviceC calls: https://www.microsoft.com, https://serviceA, https://serviceB, https://serviceC   
```

**Review by ckittel — APPROVED** [review]
Just a couple nits/wording changes. Take/leave what you need.

File: README.md (hunk 1 above)
Comment (v-fearam) [reply README.md]: I changed to 8080, but I didn't added a environment variable.

File: README.md (hunk 2 above)
Comment (v-fearam) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (v-fearam) [reply README.md]: done

File: SimpleChainApi/SimpleChainApi/Dockerfile (hunk 4 above)
Comment (v-fearam) [reply SimpleChainApi/SimpleChainApi/Dockerfile]: I changed, I hadn't time to do a deploy to test it.

File: SimpleChainApi/SimpleChainApi/Pages/Shared/_Layout.cshtml (hunk 5 above)
Comment (v-fearam) [reply SimpleChainApi/SimpleChainApi/Pages/Shared/_Layout.cshtml]: I didn't create this file. I don't know. I deleted the <style> section

File: SimpleChainApi/SimpleChainApi/wwwroot/css/site.css (hunk 6 above)
Comment (v-fearam) [reply SimpleChainApi/SimpleChainApi/wwwroot/css/site.css]: The same. I didn't write this file. I deleted that values.

File: README.md (hunk 7 above)
Comment (v-fearam) [reply README.md]: done

File: README.md (hunk 8 above)
Comment (v-fearam) [reply README.md]: done

File: README.md (hunk 9 above)
Comment (v-fearam) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (ckittel) [reply README.md]: Any reason not to?  Just going to assume that the container's default environment variable for the same will be fine?

Comment (v-fearam) [reply README.md]: I'm just worry for the lack of time to test. Tomorrow I need to ping Rick to start something else. If I can't test the new configuration I will keep the one that I'm sure is working and do it in the future.

Comment (v-fearam) [reply README.md]: done, tested

File: SimpleChainApi/SimpleChainApi/Dockerfile (hunk 4 above)
Comment (v-fearam) [reply SimpleChainApi/SimpleChainApi/Dockerfile]: done, tested

**Pull request merged**